contributingGuides/TS_STYLE.md

```diff
@@ -509,6 +511,91 @@ type Foo = {
   } satisfies Record<string, ViewStyle>;
   ```
 
+  <a name="hooks-instead-of-hocs"></a><a name="1.20"></a>
+
+- [1.20](#hooks-instead-of-hocs) **Hooks instead of HOCs**: Replace HOCs usage with Hooks whenever possible.
+  
+  > Why? Hooks are easier to use (can be used inside the function component), and don't need nesting or `compose` when exporting the component. It also allows us to remove `compose` completely in some components since it has been bringing up some issues with TypeScript. Read the [`compose` usage](#compose-usage) section for further information about the TypeScript issues with `compose`.
+
+  ```tsx
+  // BAD
+  type ComponentOnyxProps = {
+      session: OnyxEntry<Session>;
+  };
+
+  type ComponentProps = WindowDimensionsProps &
+      WithLocalizeProps &
+      ComponentOnyxProps & {
+          someProp: string;
+      };
+
+  function Component({windowWidth, windowHeight, translate, session, someProp}: ComponentProps) {
+      // component's code
+  }
+
+  export default compose(
+      withWindowDimensions,
+      withLocalize,
+      withOnyx<ComponentProps, ComponentOnyxProps>({
+          session: {
+              key: ONYXKEYS.SESSION,
+          },
+      }),
+  )(Component);
+
+  // GOOD
+  type ComponentOnyxProps = {
+      session: OnyxEntry<Session>;
+  };
+
+  type ComponentProps = ComponentOnyxProps & {
+      someProp: string;
+  };
+
+  function Component({session, someProp}: ComponentProps) {
+      const {windowWidth, windowHeight} = useWindowDimensions();
+      const {translate} = useLocalize();
+      // component's code
+  }
+
+  // There is no hook alternative for withOnyx yet.
+  export default withOnyx<ComponentProps, ComponentOnyxProps>({
+      session: {
+          key: ONYXKEYS.SESSION,
+      },
+  })(Component);
+  ```
+
+  <a name="compose-usage"></a><a name="1.21"></a>
+
+- [1.21](#compose-usage) **`compose` usage**: Avoid the usage of `compose` function to compose HOCs in TypeScript files. Use nesting instead.
+  
+  > Why? `compose` function doesn't work well with TypeScript when dealing with several HOCs being used in a component, many times resulting in wrong types and errors. Instead, nesting can be used to allow a seamless use of multiple HOCs and result in a correct return type of the compoment. Also, you can use [hooks instead of HOCs](#hooks-instead-of-hocs) whenever possible to minimize or even remove the need of HOCs in the component.
```

Comment:
Yuk! I'm all for hooks instead of HOCs, but instead of nesting HOCs like this would it be possible to fix compose to propagate props correctly?

Reply:
I think Onyx HOC is the only one that we need to use, right? For other HOCs, we have already migrated to hooks? So probably we don't have to worry about `compose` anymore if Onyx HOC is the only HOC we use?

Reply:
Good point 👍🏼

Reply:
@roryabraham Unfortunately, we tried different typings for `compose` but TS fails to infer the types in most cases, HOCs are just to complicated (especially `withOnyx`).

Reply:
> I think Onyx HOC is the only one that we need to use, right? For other HOCs, we have already migrated to hooks? So probably we don't have to worry about `compose` anymore if Onyx HOC is the only HOC we use?

Just so you know there are other HOCs that are built on top of `withOnyx`, such as `withCurrentUserPersonalDetails` that are used sometimes. However for most cases it's just `withOnyx`.

Reply:
Yes, so we will probably have only a few cases where we need to use HOC nesting, as most of the HOCs already has a equivalent Hook.